sklearn/impute/tests/test_impute.py

```diff
@@ -971,6 +971,59 @@ def test_iterative_imputer_catch_warning():
     assert not np.any(np.isnan(X_fill))
 
 
+@pytest.mark.parametrize("min_value, max_value",
+                         [(0, 100), (None, None), (-np.inf, np.inf),
+                          ((-5, 5, 10), (100, 200, 300)),
+                          ((-5, None, 10), (100, 200, None))])
+def test_iterative_imputer_min_max_completion(min_value, max_value):
+    # Test for issue #16349 which adds the capability
+    # to specify different min and max values for all features.
+    X = np.random.random((10, 3))
+    n_features = X.shape[1]
+    imputer = IterativeImputer(min_value=min_value, max_value=max_value)
+    imputer.fit(X)
+    # Check if min_value and max_value have been cast to np arrays of
+    # shape = n_features
+    assert (isinstance(imputer._min_value, np.ndarray) and
+            (imputer._min_value.shape[0] == n_features))
+    assert (isinstance(imputer._max_value, np.ndarray) and
+            (imputer._max_value.shape[0] == n_features))
+    # Make a dict with the correct values that each pair
+    # of min and max values should be cast to
+    correct_vals_dict = (
+        {(0, 100): np.array([[0] * n_features,
```

Comment:
Usually these values would be parameters to the test. Any reason not to?

Reply:
Committed changes to the test resolving this